compiler/rustc_llvm/llvm-wrapper/RustWrapper.cpp

```diff
@@ -1913,3 +1913,9 @@ extern "C" void LLVMRustContextConfigureDiagnosticHandler(
   unwrap(C)->setDiagnosticHandler(std::make_unique<RustDiagnosticHandler>(
       DiagnosticHandlerCallback, DiagnosticHandlerContext, RemarkAllPasses, Passes));
 }
+
+extern "C" void LLVMRustGetMangledName(LLVMValueRef V, RustStringRef Str) {
+  RawRustStringOstream OS(Str);
+  GlobalValue *GV = unwrap<GlobalValue>(V);
+  Mangler().getNameWithPrefix(OS, GV, true);
+}
```

Comment:
I wonder if long-term we could implement this on the `rustc` side by getting the "mangling" flavor out of the `datalayout` string.

(Also, I wish both high-level mangling and this lower-level concept weren't both called "mangling" ugh)

Reply:
Yeah, I was referring to the `m:<mangling>` field specifically. Though, looking at [the docs](https://llvm.org/docs/LangRef.html#langref-datalayout):
> * `x`: Windows x86 COFF mangling: Private symbols get the usual prefix. Regular C symbols get a `_` prefix. Functions with `__stdcall`, `__fastcall`, and `__vectorcall` have custom mangling that appends `@N` where `N` is the number of bytes used to pass parameters. C++ symbols starting with `?` are not mangled in any way.

Yikes, that's a pretty hefty ask. Even if we generated it ourselves we'd probably still want to assert that we got it right by asking the backend what *it* would generate based on its own knowledge.

Reply:
> I wonder if long-term we could implement this on the `rustc` side by getting the "mangling" flavor out of the `datalayout` string.

LLVM seems to be moving in the direction of moving the responsibility for low-level mangling to the front-end: https://github.com/llvm/llvm-project/issues/54090